models/maat/R/model2netcdf.MAAT.R

```diff
@@ -95,21 +100,36 @@ model2netcdf.MAAT <- function(outdir, sitelat = -999, sitelon = -999, start_date
     
     ### Find/replace missing and convert outputs to standardized BETYdb units
     output[[3]] <- ifelse(output[[3]] == -999, -999, 
-                          misc.convert(output[[3]],
+                          PEcAn.utils::misc.convert(output[[3]],
                                               "umol C m-2 s-1",
                                               "kg C m-2 s-1"))  # convert A/GPP to kgC/m2/s
-    output[[4]] <- ifelse(output[[4]] == "Inf", -999, 
-                          misc.convert(output[[4]], 
+    output[[4]] <- ifelse(output[[4]] == -999, -999, 
+                          PEcAn.utils::misc.convert(output[[4]],
+                                       "umol C m-2 s-1",
+                                       "kg C m-2 s-1"))  # convert leaf resp to kgC/m2/s
+    output[[5]][output[[5]]=="Inf"] <- -999
+    output[[5]][output[[5]]=="-Inf"] <- -999
+    output[[5]] <- ifelse(output[[5]] == -999, -999, 
+                          PEcAn.utils::misc.convert(output[[5]], 
                                               "mol H2O m-2 s-1",
                                               "kg H2O m-2 s-1"))  # stomatal_conductance in kg H2O m2 s1
-    
-    dims <- list(lon = lon, lat = lat, time = t)
-    
+
     ### Put output into netCDF format
+    dims <- list(lon = lon, lat = lat, time = t) # set dims for netCDF file
+    
     nc_var       <- list()
-    nc_var[[1]]  <- PEcAn.utils::to_ncvar("Year", dims)
-    nc_var[[3]]  <- PEcAn.utils::to_ncvar("GPP", dims)
-    nc_var[[4]]  <- PEcAn.utils::to_ncvar("stomatal_conductance", dims)
+    #nc_var[[1]]  <- PEcAn.utils::to_ncvar("Year", dims)
+    #nc_var[[2]]  <- PEcAn.utils::to_ncvar("FracJulianDay", dims)
+    #nc_var[[3]]  <- PEcAn.utils::to_ncvar("GPP", dims)
+    #nc_var[[4]]  <- PEcAn.utils::to_ncvar("leaf_respiration", dims)
+    #nc_var[[5]]  <- PEcAn.utils::to_ncvar("stomatal_conductance", dims)
+    
+    # temporarily reverting back to this before overhauling to use newer to_ncvar. Currently does not allow for all these vars
+    nc_var[[1]]  <- PEcAn.utils::mstmipvar("Year", lat, lon, t, NA)
```

Comment:
Instead of switching to `PEcAn.utils::to_ncvar` I believe we're going to use `ncdf4::ncvar_def` in all model wrapper functions to make it clear within the code what the dimensions and units. @mdietze

Reply:
OK, I will look at making that change